README.md

```diff
@@ -0,0 +1,89 @@
+# it-multistream-select
+
+## "Dialer"
+
+```js
+const pipe = require('it-pipe')
+const MSS = require('it-multistream-select')
+const Mplex = require('it-mplex') // https://github.com/alanshaw/it-mplex
+
+const muxer = new Mplex()
+const muxedStream = muxer.newStream()
+
+// MSS.select(stream, protocol(s))
+// Select from one of the passed protocols (in priority order)
+// Returns selected stream and protocol
+const { stream: dhtStream, protocol } = await MSS.select(muxedStream, [
+  // This might just be different versions of DHT, but could be different impls
+  '/ipfs-dht/2.0.0', // Most of the time this will probably just be one item.
+  '/ipfs-dht/1.0.0'
+])
+
+// Typically this stream will be passed back to the caller of libp2p.dialProtocol
+//
+// ...it might then do something like this:
+// try {
+//   await pipe(
+//     [Buffer.from('Some DHT data')]
+//     dhtStream,
+//     async source => {
+//       for await (const chunk of source)
+//         // DHT response data
+//     }
+//   )
+// } catch (err) {
+//   // Error in stream
+// }
+```
+
+- There's no need to separate the handshake and protocol selection. We never need to handshake and wait before selecting (right?)
```

Comment:
Correct. I'm not sure if there was an original intend to this api (@daviddias?), but we don't use these separately, so having the logic internally would be great.